src/processor/operator/persistent/reader/reader_bind_utils.cpp

```diff
@@ -23,10 +24,13 @@ void ReaderBindUtils::validateColumnTypes(const std::vector<std::string>& column
     const std::vector<common::LogicalType>& detectedColumnTypes) {
     KU_ASSERT(expectedColumnTypes.size() == detectedColumnTypes.size());
     for (auto i = 0u; i < expectedColumnTypes.size(); ++i) {
-        if (expectedColumnTypes[i] != detectedColumnTypes[i]) {
-            throw common::BinderException(common::stringFormat(
-                "Column `{}` type mismatch. Expected {} but got {}.", columnNames[i],
-                expectedColumnTypes[i].toString(), detectedColumnTypes[i].toString()));
+        if (!function::CastFunction::hasImplicitCast(expectedColumnTypes[i],
```

Comment:
I doubt this is expected. When there are multiple parquet files, we do expect their column to have the same data type because the current implementation will first scan from all parquet files and then cast.

To support what u have in mind, we need to cast inside each parquet.

Reply:
The reason I changed this is because previously, a test was giving an error because `COPY table FROM csv` failed the check in this part of the code. I think I'll disable sniffing if we're expecting types.